### ISSUE TITLE
md4all-AD model

### ISSUE THREAD

**Issue by Amor-ZYF** [md4all/md4all#20]
Sorry to bother you again! I want to implement and test your md4all-AD model on RobotCar, but in the folder config there is only train_baseline_robotcar.yaml, and in the paper it is said that the baseline is set under the situation that the translation ratio x=0. So what changes should be applied to the code if I want to implement the model md4all-AD? Thanks a lot for your reply!

**Comment by morbi25** [comment]
Hi @Amor-ZYF,

Sorry for the late reply and thanks for your question!

As described in Section 3.1.2 of our paper, md4all-AD enables us to train a baseline model to estimate depth in "easy" and adverse conditions by employing the same losses as the baseline model itself. To achieve this, we first translate our "easy" samples to adverse conditions using a pre-trained image-to-image translation network. Then, we train our model (monodepth2 + velocity loss), giving a mix of "easy" and adverse samples as input but computing the standard losses (photometric and smoothness loss) only on the "easy" samples. This means if we give the model an easy sample as input, we can directly compute the standard losses on it; however, if we give the model an input image translated to adverse conditions, we use the original corresponding easy sample to compute the standard losses. Please note, in contrast to md4all-DD, we do **not** have to train an additional baseline teacher model for knowledge distillation beforehand. :)

Therefore, you can create the configuration of md4all-AD from the [train_md4allDDa_robotcar.yaml](https://github.com/md4all/md4all/blob/main/config/train_md4allDDa_robotcar.yaml) applying the following modifications:
- Delete the DAYTIME_TRANSLATION_TEACHER_PATH parameter, as we are not using a pre-trained depth estimation model for knowledge distillation
- Delete the whole LOSS configuration section, which disables the photometric, smoothness, and velocity loss (by setting their weights to 0) and enables the knowledge distillation loss, because we would like to apply the original losses used by our baseline as written above.

I hope these steps help you! Have a nice week! :)

**Comment by Amor-ZYF** [comment]
Thank you for your reply! It's clear now! I'm closing this :)
